src/main.ts

```diff
@@ -975,7 +983,19 @@ function generateService(
     maybeAddComment(info, text => (requestFn = requestFn.addJavadoc(text)));
 
     requestFn = requestFn.addParameter('request', requestType(typeMap, methodDesc));
-    requestFn = requestFn.returns(responsePromise(typeMap, methodDesc));
+
+    // Use metadata as last argument for interface only configuration
+    if (!options.outputClientImpl && !options.outputEncodeMethods && !options.outputJsonMethods && options.addGrpcMetadata) {
```

Comment:
I'm pretty sure it's fine, but I simplified this to `if (options.addGrpcMetadata)` when I merged to master...

Hm, in retrospect, `outputEncodeMethods` and `outputJsonMethods` really shouldn't matter for adding the grpc metadata, although `outputClientImpl` would probably break b/c it'll be outputting impls that don't know about the metadata.

I think I'll keep this simplification as is and then push that out (eventually/maybe) to a config option check that addGrpcMetadata and outoutClientImpl are not compatible.